coderd/database/migrations/000351_remove_oauth2_callback_url.up.sql

```diff
@@ -0,0 +1,28 @@
+-- Migrate from callback_url to redirect_uris as source of truth for OAuth2 apps
+-- RFC 6749 compliance: use array of redirect URIs instead of single callback URL
+
+-- Populate redirect_uris from callback_url where redirect_uris is empty or NULL.
+-- NULLIF is used to treat empty strings in callback_url as NULL.
+-- If callback_url is NULL or empty, this will result in redirect_uris
+-- being an array with a single NULL element. This is preferable to an empty
+-- array as it will pass a CHECK for array length > 0, enforcing that all
+-- apps have at least one URI entry, even if it's null.
```

Comment:
Can't we make the `CHECK` check for the cardinality OR the "nullness"?

Reply:
I think I'm doing that in the client_credentials PR up in the stack, but I would need to double-check.

Reply:
I'm sorry. I just reread the comment. I'd rather not introduce a possible null here and perform a clean migration from callback_url to redirect_uris, then make sure that all of them get carried over.

The constraint for `redirect_uris` will be loosened in the PR introducing client_credentials. We'll then allow setting an empty redirect_uris array but not a null one.